nixos/modules/services/web-apps/gotosocial.nix

```diff
@@ -0,0 +1,176 @@
+        Contents of the GoToSocial YAML config.
+        Please refer to the
+        [documentation](https://docs.gotosocial.org/en/latest/configuration/)
+        and
+        [example config](https://github.com/superseriousbusiness/gotosocial/blob/main/example/config.yaml).
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.gotosocial.configFile = lib.mkDefault (settingsFormat.generate "config.yml" cfg.settings);
+    services.gotosocial.settings = (lib.mapAttrs (name: lib.mkDefault) (
+      defaultSettings // {
+        web-asset-base-dir = "${cfg.package}/share/gotosocial/web/assets/";
+        web-template-base-dir = "${cfg.package}/share/gotosocial/web/template/";
+      }
+    )) // (lib.optionalAttrs cfg.setupPostgresqlDB {
+      db-type = "postgres";
+      db-address = "/run/postgresql";
+      db-database = "gotosocial";
+      db-user = "gotosocial";
+    });
+
+    networking.firewall = lib.mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.settings.port ];
+    };
+
+    environment.systemPackages = [ gotosocial-admin ];
+
+    services.postgresql = lib.mkIf cfg.setupPostgresqlDB {
+      enable = true;
+      ensureDatabases = [ "gotosocial" ];
+      ensureUsers = [
+        {
+          name = "gotosocial";
+          ensurePermissions = {
+            "DATABASE gotosocial" = "ALL PRIVILEGES";
+          };
+        }
+      ];
+    };
+
+    systemd.services.gotosocial = {
+      description = "ActivityPub social network server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ]
+        ++ lib.optional cfg.setupPostgresqlDB "postgresql.service";
+      requires = lib.optional cfg.setupPostgresqlDB "postgresql.service";
+      restartTriggers = [ cfg.configFile ];
+
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/gotosocial server start --config-path ${cfg.configFile}";
+        Restart = "on-failure";
+        DynamicUser = true;
+        Group = "gotosocial";
+        User = "gotosocial";
+        StateDirectory = "gotosocial";
+        WorkingDirectory = "/var/lib/gotosocial";
+
+        # Security options:
+        # Based on https://github.com/superseriousbusiness/gotosocial/blob/v0.8.1/example/gotosocial.service
+        AmbientCapabilities = lib.optional (cfg.settings.port < 1024) "CAP_NET_BIND_SERVICE";
+        CapabilityBoundingSet = [
+          "CAP_RAWIO"
+          "CAP_MKNOD"
+          "CAP_AUDIT_CONTROL"
+          "CAP_AUDIT_READ"
+          "CAP_AUDIT_WRITE"
+          "CAP_SYS_BOOT"
+          "CAP_SYS_TIME"
+          "CAP_SYS_MODULE"
+          "CAP_SYS_PACCT"
+          "CAP_LEASE"
+          "CAP_LINUX_IMMUTABLE"
+          "CAP_IPC_LOCK"
+          "CAP_BLOCK_SUSPEND"
+          "CAP_WAKE_ALARM"
+          "CAP_SYS_TTY_CONFIG"
+          "CAP_MAC_ADMIN"
+          "CAP_MAC_OVERRIDE"
+          "CAP_NET_ADMIN"
+          "CAP_NET_BROADCAST"
+          "CAP_NET_RAW"
+          "CAP_SYS_ADMIN"
+          "CAP_SYS_PTRACE"
```

Comment:
Why does the program need all of those caps?

Reply:
Removed.